bool initialized_
float32 data